test/others/criu-ns/run.py

```diff
@@ -0,0 +1,185 @@
+#!/usr/bin/env python
+
+import fcntl
+import os
+import pathlib
+import pty
+import shutil
+import subprocess
+import sys
+import termios
+import time
+
+
+CRIU_BIN = "../../../criu/criu"
+CRIU_NS = "../../../scripts/criu-ns"
+IMG_DIR = "dumpdir"
+PIDFILE = "pidfile"
+
+
+def check_dumpdir(path=IMG_DIR):
+    if os.path.isdir(path):
+        shutil.rmtree(path)
+    os.mkdir(path, 0o755)
+
+
+def create_pty():
+    fd_m, fd_s = pty.openpty()
+    return (os.fdopen(fd_m, "wb"), os.fdopen(fd_s, "wb"))
+
+
+def create_isolated_dumpee():
+    pathlib.Path("running").touch()
+    fd_m, fd_s = create_pty()
+    pid = os.fork()
+    if pid == 0:
+        os.setsid()
+        os.dup2(fd_s.fileno(), 0)
+        os.dup2(fd_s.fileno(), 1)
+        os.dup2(fd_s.fileno(), 2)
+        fcntl.ioctl(fd_s.fileno(), termios.TIOCSCTTY, 1)
+        while True:
+            if not os.access("running", os.F_OK):
+                sys.exit(0)
+            time.sleep(1)
+    fd_m.close()
+    return (pid, fd_s)
+
+
+def criu_ns_dump(pid, shell_job=False):
+    if shell_job:
+        cmd = [CRIU_NS, "dump", "-D", IMG_DIR, "-v4",
```

Comment:
> when all test are succeeding there will be nothing useful (since we are only printing the log file when there is an error) in the CI log regarding this test script. Wouldn't that be confusing as well

There are many tests running in our CI, and in the case of a test failure, we try to understand the reason for the failure by looking at the logs. However, the logs generated by other tests that were successful are usually not very helpful.

Reply:
Understood, thanks! Will implement the recommended changes.